aten/src/ATen/core/ivalue.cpp

```diff
@@ -287,12 +287,15 @@ std::ostream& printMaybeAnnotatedList(
     std::ostream& out,
     const IValue& the_list,
     IValueFormatter formatter) {
-  if (the_list.toListRef().size() == 0) {
-    out << "annotate(" << the_list.type()->python_str() << ", [])";
+
+  if (the_list.toListRef().size() == 0 || elementTypeCanBeInferredFromMembers(the_list.type())) {
```

Comment:
at a very quick glance, shouldn't this be `!elementTypeCanBeInferredFromMembers`, like we should annotate when we can **not** infer the type from members?

Reply:
good catch. that's where test failures are coming from now